keras_cv/layers/preprocessing/resizing.py

```diff
@@ -143,24 +154,37 @@ def _augment(self, inputs):
             )
             inputs["bounding_boxes"] = outputs["bounding_boxes"]
 
+        if segmentation_maps is not None:
+            segmentation_maps = tf.squeeze(outputs["segmentation_maps"], axis=0)
+            inputs["segmentation_maps"] = segmentation_maps
+
         return inputs
 
     def _resize_with_distortion(self, inputs):
         images = inputs.get("images", None)
+        segmentation_maps = inputs.get("segmentation_maps", None)
 
         size = [self.height, self.width]
         images = tf.image.resize(
             images, size=size, method=self._interpolation_method
         )
         images = tf.cast(images, self.compute_dtype)
 
+        if segmentation_maps is not None:
+            segmentation_maps = tf.image.resize(
+                segmentation_maps, size=size, method=self._interpolation_method
+            )
```

Comment:
I think we should have a way to input different resizing methods of the image and the segmentation masks.

WDYT? @ianstenbit

Reply:
Usually segmentation masks are resized with nearest neighbor interpolation in practice. Using say bilinear interpolation ruin the masks.

Reply:
Yes, this should be resized with nearest neighbor. Numerical interpolation is not safe for masks unless they're one-hot encoded. (See RandomRotation for an example of where we had to deal with this).

For resizing I think it's probably reasonable to just always use nearest neighbor, regardless of whether masks are one-hot encoded.